homeassistant/components/squeezebox/media_player.py

```diff
@@ -451,6 +457,24 @@ async def async_play_media(
             )
             media_id = play_item.url
 
+        if announce:
+            if media_type not in MediaType.MUSIC:
+                raise ValueError(
+                    "Announcements must have media type of 'music'.  Playlists are not supported"
+                )
+            extra = kwargs.get(ATTR_MEDIA_EXTRA, {})
+            cmd = "announce"
+            self._player.set_announce_volume(
+                int(extra["announce_volume"])
+                if extra.get("announce_volume", None)
+                else None
+            )
+            self._player.set_announce_timeout(
+                int(extra["announce_timeout"])
+                if extra.get("announce_timeout", None)
```

Comment:
```suggestion
                if extra.get("announce_volume")
                else None
            )
            self._player.set_announce_timeout(
                int(extra["announce_timeout"])
                if extra.get("announce_timeout")
```
Do note, if both items are `0`, this is considered falsy. So if I set an announce timeout of 0, it will do `None`

Reply:
Interesting - thanks for the note.  Likely the right behaviour though in both cases I think.  A volume of zero for an announcement or a timeout of zero should likely do nothing anyway.  Is there a way of distinguishing between no value submitted and zero submitted that we could use and raise a valuerror, or just ignore it.

Reply:
if extra.get("...") is not None

But what does a timeout of 0 mean?

Reply:
I know Sonos has a built in "announce" function, but sadly we don't have that on squeezebox, so I have to save the state, play the item (with possibly lowered or louder volume) and the restore the state.

Reply:
OK - have added some checks for 0 in both cases and raised errors, and added the tests for each